tests/test_trial.py

```diff
@@ -1319,3 +1319,25 @@ def objective(trial: optuna.trial.Trial) -> Tuple[float, float]:
         )
 
     study.optimize(objective, n_trials=10)
+
+
+def test_report_with_multi_objectives() -> None:
+    study = optuna.create_study(directions=["maximize", "maximize"])
+    trial = Trial(study, study._storage.create_new_trial(study._study_id))
+
+    # Try to report values but nothing happens.
+    trial.report(1.0, 0)
+    trial.report([1.0, 1.0], 1)  # type: ignore
```

Comment:
Are we intentionally not checking the warnings?

Reply:
I'm not sure we should test the warnings of `_logger.warn(...)` and how to do it. Do you have any idea?

Reply:
I tested it by using not `logging.warn` but `warnings.warn`. See https://github.com/optuna/optuna/pull/2135#discussion_r548322622.